file.go

```diff
@@ -134,37 +149,20 @@ func ioCompletionProcessor(h syscall.Handle) {
 
 // asyncIo processes the return value from ReadFile or WriteFile, blocking until
 // the operation has actually completed.
-func (f *win32File) asyncIo(c *ioOperation, deadline time.Time, bytes uint32, err error) (int, error) {
+func (f *win32File) asyncIo(c *ioOperation, timeout timeoutChan, bytes uint32, err error) (int, error) {
```

Comment:
>timeout timeoutChan [](start = 44, length = 19)

better to make this a bool and get the channel just before the select. this way, you can grab a mutex to get the channel, and you can avoid taking the mutex for IOs that complete synchronously